pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -678,6 +678,11 @@ func (ctrl *Controller) syncImageConfig(key string) error {
 		}
 	}
 
+	// append the internalRegistryHostname to allowedRegistries list
+	allowedRegs := imgcfg.Spec.RegistrySources.AllowedRegistries
+	if len(imgcfg.Status.InternalRegistryHostname) > 0 {
+		allowedRegs = append(allowedRegs, imgcfg.Status.InternalRegistryHostname)
+	}
```

Comment:
Please organize this code to only have one copy of the logic; at least a helper function like `getValidBlockedRegs`, or ideally in a way that is impossible to bypass by mistake (_maybe_ pass all of `imgcfg` into `registriesConfigIgnition` and move the log somewhere inside that call subtree).